backend/src/test/java/it/com/crm/api/v1/CarControllerIntegrationTest.java

```diff
@@ -0,0 +1,88 @@
+package it.com.crm.api.v1;
+
+import com.crm.App;
+import com.crm.model.db.CarEntity;
+import com.crm.repository.CarRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest(classes = App.class)
+@AutoConfigureMockMvc
+@WithMockUser
+@Transactional
+class CarControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private CarRepository carRepository;
+
+    @Test
+    void shouldResponseEntityHasPageNumberEqualTo1() throws Exception {
+        //when
+        mvc.perform(get("/cars?page=1"))
+                .andDo(print())
+                //then
+                .andExpect(status().is(200))
+                .andExpect(jsonPath("$.pageable.pageNumber", is(1)));
+    }
+
+    @Test
+    void shouldResponseEntityHasPageNumberEqualTo2WhenParamSizeEqualTo1() throws Exception {
+        //given
+        CarEntity carEntity = new CarEntity();
+        carEntity.setBrand("testBrand");
+        carRepository.save(carEntity);
```

Comment:
if code will be deployed on prod server, when you run integration test, this data will be stored in production database. I think its better to use mock for repository or configure h2 database only for integration test purpose.

Reply:
H2 is added